pkg/helpers/newapp/app/imageref.go

```diff
@@ -230,18 +236,45 @@ func (r *ImageRef) SuggestName() (string, bool) {
 	if r == nil {
 		return "", false
 	}
+	var name string
 	// if ImageStream already exists, its name should
 	// have higher preference than object name
-	if r.Stream != nil {
-		return r.Stream.Name, true
+	switch {
+	case r.Stream != nil:
+		name = r.Stream.Name
+	case len(r.ObjectName) > 0:
+		name = r.ObjectName
+	case len(r.Reference.Name) > 0:
+		name = r.Reference.Name
 	}
-	if len(r.ObjectName) > 0 {
-		return r.ObjectName, true
+
+	name = toValidImageRefName(name)
+
+	return name, len(name) > 0
+}
+
+// checkNameIsValidSubdomain checks that ImageRef suggested name is a valid
```

Comment:
@alicerum missed the update to the godoc here.

Reply:
oops

Reply:
thanks for noticing!